### ISSUE TITLE
Nullable upsert field can't be null

### ISSUE THREAD

**Issue by Linloir** [medz/prisma-dart#211]
I set a nullable field in my schema as follows:

```prisma
model foo {
  // ...
  deleteAt    DateTime?
  // ...
}
```

The generation goes well and an upsert interface containing a nullable `deleteAt` field is generated as expected.

However something is going wrong inside the parsing engine maybe, and I get an error whenever I try to set that field to null (in cases that I didn't delete that item).

The invoke function is kinda like:

```dart
final foo = await PrismaProvider.client.foo.upsert(
  where: FooWhereUniqueInput(
    id: baz,
  ),
  create: FooCreateInput(
    deleteAt: null
  ),
  update: FooUpdateInput(
    deleteAt: NullableDateTimeFieldUpdateOperationsInput(
      set: deleted ? deletedTime : null
    ),
  ),
);
```

The error message is:

PrismaRequestException (PrismaRequestException: Unable to match input value to any allowed input type for the field. Parse errors: [Unable to match input value to any allowed input type for the field. Parse errors: [Invalid argument type. `deleteAt` should be of any of the following types: `DateTime`, Some fields are missing: Expected exactly one field to be present, got 0., Invalid argument type. `deleteAt` should be of any of the following types: `Null`], Unable to match input value to any allowed input type for the field. Parse errors: [Invalid argument type. `deleteAt` should be of any of the following types: `DateTime`, Some fields are missing: Expected exactly one field to be present, got 0., Invalid argument type. `deleteAt` should be of any of the following types: `Null`]])

**Comment by Linloir** [comment]
Tried changing field type to `BigInt` and the problem still remains

**Comment by vfiruz97** [comment]
+1

**Comment by medz** [comment]
I have good news, the v4 version is almost finished, it reintroduces the PrismaUnion type (using data type nesting to simulate union-types) and the PrismaNull type, which makes this problem solved, please read #290 for progress
